pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisStreamMetadataProvider.java

```diff
@@ -83,7 +91,7 @@ public long fetchPartitionOffset(@Nonnull OffsetCriteria offsetCriteria, long ti
     List<PartitionGroupInfo> newPartitionGroupInfos = new ArrayList<>();
 
     Map<String, Shard> shardIdToShardMap =
-        _kinesisConnectionHandler.getShards().stream().collect(Collectors.toMap(Shard::shardId, s -> s));
+        _kinesisConnectionHandler.getShards().stream().collect(Collectors.toMap(Shard::shardId, s -> s, (s1, s2) -> s1));
```

Comment:
why would a collision happen here?

Reply:
I also wasn't able to figure this out but while running a collision was definitely happening and error was being thrown.